src/expr/core/src/sig/mod.rs

```diff
@@ -141,41 +161,17 @@ pub struct FuncSign {
     /// The return type.
     pub ret_type: SigDataType,
 
-    /// A function to build the scalar function.
-    pub build_scalar: Option<ScalarFunctionBuilder>,
-
-    /// A function to build the table function.
-    pub build_table: Option<TableFunctionBuilder>,
-
-    /// A function to build the aggregate function (retractable version).
-    pub build_aggregate_retractable: Option<AggregateFunctionBuilder>,
-
-    /// A function to build the aggregate function (append-only version).
-    pub build_aggregate_append_only: Option<AggregateFunctionBuilder>,
+    /// A function to build the expression.
+    pub build: FuncBuilder,
 
     /// A function to infer the return type from argument types.
     pub type_infer: fn(args: &[DataType]) -> Result<DataType>,
 
     /// Whether the function is deprecated and should not be used in the frontend.
     /// For backward compatibility, it is still available in the backend.
     pub deprecated: bool,
```

Comment:
Is it better to have another field representing `internal`, since internal functions may also be _deprecated_. Not a strong opinion though.

Reply:
`internal` is just an alias of `deprecated`. They are the same from the view of signature. 🤔